src/client/gui/lib/update_available.dart

```diff
@@ -15,14 +17,39 @@ class UpdateNotifier extends Notifier<UpdateInfo> {
 
   void set(UpdateInfo updateInfo) {
     if (updateInfo.version.isBlank) return;
+
     final updateNotificationExists = ref.read(notificationsProvider).any((n) {
       return n is UpdateAvailableNotification && n.updateInfo == updateInfo;
     });
     if (updateNotificationExists) return;
-    ref
-        .read(notificationsProvider.notifier)
-        .add(UpdateAvailableNotification(updateInfo));
+
+    // Existing in-app notification
+    ref.read(notificationsProvider.notifier).add(UpdateAvailableNotification(updateInfo));
+
+    // Update the state
     state = updateInfo;
+
+    // Create and show a local notification
+    _showLocalNotification(updateInfo);
+  }
+
+  void _showLocalNotification(UpdateInfo updateInfo) {
+
+    if (Platform.isLinux) {
```

Comment:
We generally try to avoid using `Platform.is<OS>` directly. Take a look at the `MpPlatform` class. You can add a bool getter there and override it in each derived class. Then you can use here the `mpPlatform` singleton to check the condition. It is a little more cumbersome, but it keeps all platform specific behavior centralized.